crates/biome_js_analyze/src/lint/nursery/use_trim_start_end.rs

```diff
@@ -0,0 +1,181 @@
+            "Use "{replaced_member_name}" instead of "{member_name}"."
+        }
+        .to_owned();
+        let note_message = {
+            markup! {
+                ""{member_name}"() is an alias for "{replaced_member_name}"."
+            }
+            .to_owned()
+        };
+
+        Some(
+            RuleDiagnostic::new(rule_category!(), state.span, diagnostic_message)
+                .note(note_message),
+        )
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let callee = node.callee().ok()?;
+        let token = generate_syntax_token(callee)?;
+
+        let member_name = state.member_name.text();
+        let replaced_member_name = state.replaced_member_name.clone();
+        let replaced_function = make::js_name(make::ident(&replaced_member_name));
+        let mut mutation = ctx.root().begin();
+        mutation.replace_element(NodeOrToken::Token(token), replaced_function.into());
+
+        Some(JsRuleAction::new(
+            ActionCategory::QuickFix,
+            ctx.metadata().applicability(),
+            markup! { "Replace "<Emphasis>{member_name}</Emphasis>" with "<Emphasis>{replaced_member_name}</Emphasis>"." }
+                .to_owned(),
+            mutation,
+        ))
+    }
+}
+
+fn generate_syntax_token(callee: AnyJsExpression) -> Option<SyntaxToken<JsLanguage>> {
+    let token = if let AnyJsExpression::JsComputedMemberExpression(expression) = callee {
+        let member = expression.member().ok()?;
+        match member {
+            AnyJsExpression::AnyJsLiteralExpression(literal) => literal.value_token().ok(),
+            AnyJsExpression::JsTemplateExpression(element) => {
+                element.elements().into_iter().find_map(|x| {
+                    x.as_js_template_chunk_element()
+                        .and_then(|chunk| chunk.template_chunk_token().ok())
+                })
+            }
+            _ => None,
+        }
+    } else if let AnyJsExpression::JsStaticMemberExpression(expression) = callee {
+        expression.member().ok()?.value_token().ok()
+    } else {
+        None
+    };
+    token
+}
+
+// Handle "'text'" and "\"text\"" and "text" cases
+fn suggested_name(text: &SyntaxToken<JsLanguage>) -> String {
+    let trimmed = text.text_trimmed();
+    let first_char = trimmed.chars().next();
+    let last_char = trimmed.chars().last();
+
+    let is_single_quoted = first_char == Some('\'') && last_char == Some('\'');
+    let is_double_quoted = first_char == Some('"') && last_char == Some('"');
+
+    let unquoted = trimmed.trim_matches(|c| c == '\'' || c == '"');
+
+    let cleaned = match unquoted {
+        "trimLeft" => "trimStart",
+        "trimRight" => "trimEnd",
+        _ => unquoted,
+    };
+
+    if is_single_quoted {
+        format!("'{}'", cleaned)
+    } else if is_double_quoted {
+        format!("\"{}\"", cleaned)
+    } else {
+        cleaned.to_string()
+    }
+}
```

Comment:
I don't think we need these helper functions.

We have some useful utility functions, for example: https://github.com/biomejs/biome/blob/219dae0ef579e104c71efd47edbf9ed1545d136e/crates/biome_js_syntax/src/expr_ext.rs#L950-L982

You can refer to https://github.com/biomejs/biome/blob/219dae0ef579e104c71efd47edbf9ed1545d136e/crates/biome_js_analyze/src/lint/complexity/use_literal_keys.rs#L79-L80 to get the idea of how to use them.